packages/client/src/arpa_reporter/views/NewUpload.vue

```diff
@@ -1,111 +1,164 @@
 <template>
-  <div class="upload">
-    <h1>Submit Workbook</h1>
+  <div>
 
-    <div>
-      <div v-if="error" class="mt-3 alert alert-danger" role="alert">
-        {{ error }}
-      </div>
-
-      <form
-        method="post"
-        encType="multipart/form-data"
-        ref="form"
-        @submit.prevent="onSubmit"
-      >
-        <div class="form-group">
-          <input
-            class="form-control-file"
-            type="file"
-            id="spreadsheet"
-            name="spreadsheet"
-            @change="changeFiles"
-            ref="files"
-          />
-        </div>
-        <div class="form-group">
-          <button
-            class="btn btn-primary"
-            type="submit"
-            :disabled="uploadDisabled"
-            @click.prevent="onSubmit"
-          >
-            {{ uploadButtonLabel }}
-          </button>
-          <a class="ml-3" href="#" @click="cancelUpload">Cancel</a>
-        </div>
-      </form>
-    </div>
+    <FormulateForm
+      name="new-upload"
+      :form-errors="formErrors"
+      v-model="values"
+      :schema="schema"
+      @submit="onSubmit"
+    >
+    <FormulateErrors
+      class="alert alert-danger"
+    />
+    </FormulateForm>
   </div>
 </template>
 
 <script>
+import Vue from 'vue';
+import VueFormulate from '@braid/vue-formulate';
 import { postForm } from '../store';
 
+Vue.use(VueFormulate, {
+  classes: {
+    prefix: 'formulate-',
+    outer: 'form-group',
+    input: (context, classes) => {
+      // set this to 'form-control' unless it's a file input
+      const inputClass = context.type === 'file' ? '' : 'form-control';
+      return [classes, inputClass].join(' ');
+    },
+    inputHasErrors: 'is-invalid',
+    help: 'form-text text-muted',
+    errors: 'list-unstyled text-danger',
+  },
+});
 export default {
-  name: 'NewUpload',
   data() {
     return {
-      error: null,
-      files: null,
-      uploading: false,
+      formErrors: [],
+      initialValues: () => ({
+        reportingPeriodId: this.$store.getters.viewPeriodID ?? '',
+        agencyId: this.$store.getters.user.agency.id,
+      }),
+
+      values: {},
```

Comment:
```suggestion
      values: this.initialValues(),
```

Instead of populating this in the `created` lifecycle hook, can we just populate it here?

Reply:
I had briefly tried this before only to find it didn't populate correctly. That said! Trying it again, I'm realizing now, in having used an arrow function, I hadn't sufficiently captured `this`, which was the true culprit!

Reply:
Ahh, nice catch!  Yeah, that would be a sneaky one.